### PR TITLE
[new release] uchar (0.0.2+dune2)

### PR DESCRIPTION
**Pull request by emillon**
Dummy but non virtual package to satisfy both opam-monorepo and opam

- Project page: <a href="https://github.com/dune-universe/uchar">https://github.com/dune-universe/uchar</a>

### CODE REVIEW

**Review by Leonidas-from-XIV — APPROVED** [review]

**Comment by NathanReb** [comment]
cc @maiste

**Comment by emillon** [comment]
`uchar.v0.0.2+dune` should require `dune < 3`. OK to do that in this PR?

**Comment by Leonidas-from-XIV** [comment]
I would say yes, similarly how this would be done in opam-repository.

File: packages/uchar/uchar.0.0.2+dune2/opam [hunk 1]
```diff
@@ -0,0 +1,33 @@
+opam-version: "2.0"
+maintainer: "Daniel Bünzli <daniel.buenzl i@erratique.ch>"
+authors: ["Daniel Bünzli <daniel.buenzl i@erratique.ch>"]
+homepage: "https://github.com/dune-universe/uchar"
+dev-repo: "git+https://github.com/dune-universe/uchar.git"
+bug-reports: "https://github.com/dune-universe/uchar/issues"
+tags: [ "text" "character" "unicode" "compatibility" "org:ocaml.org" ]
+license: "LGPL-2.1-only WITH OCaml-LGPL-linking-exception"
+depends: [ "dune" {>= "1.1"} "ocaml" {>= "4.03"} ]
```
Comment (samoht) [inline packages/uchar/uchar.0.0.2+dune2/opam:9]:
```suggestion
depends: [ "dune" {>= "1.1" & < "3.0"} "ocaml" {>= "4.03"} ]
```

**Comment by samoht** [comment]
I've added the constraint for dune.

**Comment by samoht** [comment]
Actually: what's the difference between `+dune` and `+dune2`? Why do we need a new package with the same archive?

**Comment by Leonidas-from-XIV** [comment]
@samoht We add the `+dune2` suffix when we release a new tarball with a revised/fixed/improved dune patch. That way people that have locked the respective `+dune` version before can use it while new `opam-monorepo lock` will pick the `+dune2` version.

For new releases we "reset" to `+dune` again, a bit like Debian does. It is a bit unfortunate that `+dune2` sounds like a port to dune 2.x but it isn't.

**Comment by samoht** [comment]
ha ok so you were suggesting to fix the `dune` dependency in the other `+dune` package. Got it. I'll do this.

**Comment by samoht** [comment]
I've force-pushed the changes.

**Comment by samoht** [comment]
The lint issue is on the `+dune` package. I guess we can't do much there:

```
Sys_error("dune-project: No such file or directory")
                Raised by primitive operation at Stdlib.open_in_gen in file "stdlib.ml", line 405, characters [28](https://github.com/dune-universe/opam-overlays/runs/5689488099?check_suite_focus=true#step:8:28)-54
                Called from Stdio__In_channel.with_file in file "src/in_channel.ml", line 18, characters 45-66
                Called from Duniverse_lint__Lint.dune_project_exists in file "lib/lint.ml", line 63, characters 4-2[30](https://github.com/dune-universe/opam-overlays/runs/5689488099?check_suite_focus=true#step:8:30)
                Called from Duniverse_lint__Lint.eval in file "lib/lint.ml", line 78, characters 8-18
                Called from Base__List.concat_map.aux in file "src/list.ml", line 709, characters 34-40
                Called from Dune__exe__Lint.check in file "bin/lint.ml", line 11, characters 16-[31](https://github.com/dune-universe/opam-overlays/runs/5689488099?check_suite_focus=true#step:8:31)
                Called from Dune__exe__Lint.main in file "bin/lint.ml", line 15, characters 20-28
                Called from Cmdliner_term.app.(fun) in file "cmdliner_term.ml", line 24, characters 19-24
                Called from Cmdliner_eval.run_parser in file "cmdliner_eval.ml", line [34](https://github.com/dune-universe/opam-overlays/runs/5689488099?check_suite_focus=true#step:8:34), characters 37-44
```

**Pull request merged**